lib/mainScreen/widgets/dialogs/main_factionDialog.dart

```diff
@@ -16,52 +16,55 @@ class FactionDialog extends AlertDialog {
-                      child: Container(
-                        color: Theme.of(parentContext).primaryColor,
-                        padding: EdgeInsets.all(10),
-                        child: Text(
-                          "Faction",
-                          textScaleFactor: 3.0,
-                          textAlign: TextAlign.center,
-                          style: TextStyle(
-                            color: Colors.white,
+    child: SingleChildScrollView(
+        child: Column(
+            crossAxisAlignment: CrossAxisAlignment.center,
+            mainAxisSize: MainAxisSize.min,
+            children: <Widget>[
+              Row(
+                  mainAxisSize: MainAxisSize.max,
+                  children: <Widget>[
+                    Expanded(
+                        child: Container(
+                          color: Theme.of(parentContext).primaryColor,
+                          padding: EdgeInsets.all(10),
+                          child: Text(
+                            "Faction",
+                            textScaleFactor: 3.0,
+                            textAlign: TextAlign.center,
+                            style: TextStyle(
+                              color: Colors.white,
+                            ),
                           ),
-                        ),
-                      )
-                  )
-                ]
-            ),
-            Container (
-              margin: EdgeInsets.only(top: 10, bottom: 10),
-              child: Text (
-                faction.name,
-                textScaleFactor: 2.0,
-                style: TextStyle(
-                    fontWeight: FontWeight.bold
+                        )
+                    )
+                  ]
+              ),
+              Container (
+                margin: EdgeInsets.only(top: 10, bottom: 10),
+                child: Text (
+                  faction.name,
+                  textScaleFactor: 2.0,
+                  textAlign: TextAlign.center,
+                  style: TextStyle(
+                      fontWeight: FontWeight.bold
+                  ),
                 ),
               ),
-            ),
-            Container (
-              padding: EdgeInsets.only(left: 10, right: 10),
-              child: Text(
-                faction.description,
-                textAlign: TextAlign.center,
-                style: TextStyle(
-                    fontStyle: FontStyle.italic
+              Container (
+                padding: EdgeInsets.only(left: 10, right: 10),
+                child: Text(
+                  faction.description,
+                  textAlign: TextAlign.center,
+                  style: TextStyle(
+                      fontStyle: FontStyle.italic
+                  ),
                 ),
               ),
-            ),
-            ..._getFactionModifierDisplays()
-          ]
-      )
+              ..._getFactionModifierDisplays()
+            ]
+        )
+    )
```

Comment:
This change was made in response to a small glitch on a testing user's Huawei P9. It simply wraps the existing view in a scrolling view. I'm unable to confirm if this solves the issue, but this is my best guess at a solution.